### PR TITLE
Fixed and Improved 18673 script and Fixed Imperial Spear script

### LINKED ISSUE #3240: Imperial Spear Refine Bonus Error
* **Client Date**: 
20170614

* **Server Mode**: 
Renewal

* **Description of Issue**: 
  * Result: 3% damage bonus for Vanishing Point and Canon Spear skills gained from refine level is not calculated in item script.
  * Expected Result: Should include 3% per 2 refines.
```
{ .@r = getrefine(); bonus2 bSkillAtk,"LG_CANNONSPEAR",20+(3*(.@r/2)); bonus2 bSkillAtk,"LG_BANISHINGPOINT",20+(3*(.@r/2)); },{},{}
```

### PR DESCRIPTION
**Pull request by BrOgBr**
* **Addressed Issue(s)**: 

* **Server Mode**:  Renewal

* **Description of Pull Request**: 

I'm learning to use "?", so I would like if someone can confirm if it's correct.

I used DP (kRO) to collect the infos.
https://www.divine-pride.net/database/item/18673

Fixed #3240 
Thanks @slyx88

### CODE REVIEW

File: db/re/item_db.txt [hunk 1]
```diff
@@ -9701,7 +9701,7 @@
 18670,Hankie_In_Mouth,Hankie In Mouth,4,20,,100,,1,,0,0xFFFFFFFF,63,2,1,,12,0,818,{ bonus bDex,3; bonus2 bSubRace,RC_DemiHuman,3; bonus2 bSubRace,RC_Player,3; },{},{}
 18671,Rudolf_Hairband,Rudolf Hairband,4,20,,200,,5,,0,0xFFFFFFFF,63,2,256,,30,1,836,{},{},{}
 18672,Tare_Pope,Drooping Pope,4,20,,300,,0,,0,0xFFFFFFFF,63,2,256,,1,0,817,{ bonus bMdef,1; bonus bSPrecovRate,2; bonus bUnbreakableHelm; },{},{}
-18673,Tare_Pope_,Drooping Pope,4,20,,300,,0,,1,0xFFFFFFFF,63,2,256,,1,1,817,{ bonus bMdef,7; bonus bSPrecovRate,2; bonus bUnbreakableHelm; .@r = getrefine(); if (.@r >= 7 ) { bonus bHealPower,1; } if (.@r >= 9 ) {bonus bMatkRate,7; } if (.@r >= 12 ) {bonus bMatkRate,5; } },{},{}
+18673,Tare_Pope_,Drooping Pope,4,20,,300,,0,,1,0xFFFFFFFF,63,2,256,,1,1,817,{ bonus bUnbreakableHelm; bonus bMdef,7; .@r = getrefine(); bonus bHealPower,(.@r>=7?5+(.@r-6):5); bonus bMatkRate,(.@r>=12?12:(.@r>=9?7:0)); },{},{}
```
Comment (Lemongrass3110) [inline db/re/item_db.txt]: I would put the bMatkRate into a real if to get rid of the case where you give the bonus with value 0.

Comment (Atemo) [reply db/re/item_db.txt]: From where did you get the value for `bonus bHealPower,(.@r>=7?5+(.@r-6):5);` ?

Comment (BrOgBr) [reply db/re/item_db.txt]: @Lemongrass3110 Sorry, I didn't understand what did you mean, bMatkRate is 0 when refine is less than 9.

>9제련 시 MATK + 7% 추가 증가.
12제련 시 MATK + 5% 추가 증가.

@Atemo at Divine-Pride using kRO DB, in the description, it says when refine is less than 7 HP 5%, when refine 7 or more, each refine +1% HP. (HP = Heal Power)

>힐량 5% 증가.
7제련 시부터 1제련당 힐량 1%씩 추가 증가.

Is it wrong? I can't confirm in the Aegis.

Comment (BrOgBr) [reply db/re/item_db.txt]: @Atemo Nice! Ok, I'll do this then. but the bMatkRate with value 0 is for less then +9.
I didn't know that it couldn't stay that way. I thought that using "?" would be more "clean" for the emulator reads.

Comment (Lemongrass3110) [reply db/re/item_db.txt]: Yeah what i was thinking was if you want to use short if is to make it like this:
`if (.@r>=9) { bonus bMatkRate, ( .@r >= 12 ? 12 : 7 ); }`
But dont have something give bonus with 0 :-)

**Pull request merged**